src/lib.rs

```diff
@@ -7,6 +7,16 @@ use std::collections::HashSet;
 pub mod diagnostics;
 use diagnostics::{Diagnostic, DiagnosticSpan};
 
+pub fn get_suggestions_from_json(input: &str, only: &HashSet<String>) -> Vec<Suggestion> {
```

Comment:
Should note that this expects rustc's JSON output. Cargo puts each document in a `{"message": rustc_doc}` wrapper